neuro_san/http_sidecar/http_sidecar.py

```diff
@@ -98,13 +102,24 @@ def make_app(self):
     def allow(self, agent_name) -> bool:
         return self.allowed_agents.get(agent_name, False)
 
+    def update_agents(self, agents: List[str]):
+        with self.lock:
+            # We assume all agents from "agents" dictionary are enabled:
+            for agent_name in agents:
+                self.allowed_agents[agent_name] = True
+            # All other agents are disabled:
+            for agent_name, _ in self.allowed_agents.items():
+                if agent_name not in agents:
+                    self.allowed_agents[agent_name] = False
+
```

Comment:
Update our internal dynamic collection of known agents with actual set of agents coming from some external "source of truth". This internal collection determines which requests will be handled, and which "404" rejected.

Reply:
This isn't ideal because now every single request has to go through a lock.
Thinking (perhaps foolishly) out loud: Perhaps some kind of versioning mechanism could be put in place where requests can go about their business without a lock if they have the same version as what the single-source-of-truth has.

Reply:
Locking itself is not scary if: we know that code under lock will execute fast and will not involve any additional locking logic. That's the case here.
To be honest though, this here lock is not strictly necessary since our Tornado server runs a (single) event loop.
But: I expect this to change when we get rid of gRPC layer and will have to refactor our threads/event loops structure.